repository/lib/rom/repository/class_interface.rb

```diff
@@ -113,7 +113,13 @@ def define_restricted_command_method(type, views, **opts)
             if changeset.respond_to?(:commit)
               changeset.commit
             else
-              root.command(type, **opts).public_send(view_name, *view_args).call(*input)
+              #   in case of view_args been coercible to an array we do not want to splat it
+              #   example: Date and Time objects
+              if view_args.respond_to?(:to_a)
+                root.command(type, **opts).public_send(view_name, view_args).call(*input)
```

Comment:
To avoid the reassignment you could even do the splatting in the original assignment like so:

```ruby
define_method(meth_name) do |*args|
  (*view_args), *input = args
  # ...
end
```